File: companyy/src/main/java/stock/example/companyy/entities/Order.java
package stock.example.companyy.entities;


import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.*;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;
import java.util.List;
import java.util.Map;

@Document(collection = "orders")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private Date date;

    //private List<product> products;
    private Map<Integer, Integer> products;

    private User client;

    private User Agent;

    private double tva;
    private double totalPrice;


}

File: angular/src/app/Models/Order.ts
import { Product } from "./Product";
import { User } from "./User";

export class Order {
    id!: number;
    date!: Date;
    products!: Map<number, number>; // Product with quantity
    client: User | undefined;
    agent: User | undefined;
    tva!:number;
    totalPrice!:number
  }